naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/SangriaGraphQlSchemaBuilder.scala

```diff
@@ -0,0 +1,537 @@
+import sangria.schema.EnumType
+import sangria.schema.EnumValue
+import sangria.schema.Field
+import sangria.schema.FloatType
+import sangria.schema.InputType
+import sangria.schema.IntType
+import sangria.schema.ListInputType
+import sangria.schema.ListType
+import sangria.schema.LongType
+import sangria.schema.OptionType
+import sangria.schema.OutputType
+import sangria.schema.Schema
+import sangria.schema.StringType
+import sangria.schema.UnionType
+import sangria.schema.Value
+import sangria.marshalling.FromInput._
+import sangria.schema.ObjectType
+import schema.ResourceCompanion
+
+import scala.collection.JavaConverters._
+
+class SangriaGraphQlSchemaBuilder(
+    resources: Set[Resource],
+    schemas: Map[String, RecordDataSchema])
+  extends StrictLogging {
+
+  import SangriaGraphQlSchemaBuilder._
+
+  /**
+    * Generates a GraphQL schema for the provided set of resources to this class
+    * Returns a "root" object that has one field available for each Naptime Resource provided.*
+    *
+    * @return a Sangria GraphQL Schema with all resources defined
+    */
+  def generateSchema(): Schema[SangriaGraphQlContext, DataMap] = {
+    val topLevelResourceObjects = resources.map { resource =>
+      val resourceObject = generateLookupTypeForResource(ResourceCompanion.versionedName(resource))
+      Field.apply[SangriaGraphQlContext, DataMap, DataMap, Any](
+        formatResourceTopLevelName(resource),
+        resourceObject,
+        resolve = (context: Context[SangriaGraphQlContext, DataMap]) => {
+          Value(null)
+        })
+    }
+
+    val dedupedResources = topLevelResourceObjects.groupBy(_.name).map(_._2.head).toList
+    val rootObject = ObjectType[SangriaGraphQlContext, DataMap](
+      name = "Root",
+      description = "Top-level accessor for Naptime resources",
+      fields = dedupedResources)
+    Schema(rootObject)
+  }
+
+  /**
+    * Generates an object-type for a given resource name, with each field on the merged output
+    * schema available on this object-type.
+    *
+    * @param resourceName String name of the resource (i.e. 'courses.v1')
+    * @return ObjectType for the resource
+    */
+  def generateObjectTypeForResource(
+      resourceName: String): ObjectType[SangriaGraphQlContext, DataMap] = {
+    try {
+      val resource = getResource(resourceName)
+
+      val schema = schemas.get(resource.mergedType).flatMap(Option(_)).getOrElse {
+        throw new RuntimeException(s"Cannot find schema for ${resource.mergedType}")
+      }
+
+      ObjectType[SangriaGraphQlContext, DataMap](
+        name = formatResourceName(resource),
+        fieldsFn = () => {
+          Option(schema.getFields).map(_.asScala.map { field =>
+            generateField(field, Option(schema.getNamespace).getOrElse(""))
+          }.toList).getOrElse(EMPTY_FIELDS_FALLBACK)
+        })
+    } catch {
+      case e: Throwable =>
+        logger.error(s"Could not generate object type for resource $resourceName")
```

Comment:
we should probably bring these down to warnings. AND/OR collect all of the objects that we couldn't generate types for, and combine them to a single message.